recipes-containers/kubelet/files/launch-kubelet.sh

```diff
@@ -19,24 +20,78 @@
 
 DEVICE_ID=`jq -r .deviceID /userdata/edge_gw_config/identity.json`
 if [[ $? -ne 0 ]] || [[ $DEVICE_ID == null ]]; then
-    echo "Unable to extract device ID from identity.json"
-    exit 1
+	echo "Unable to extract device ID from identity.json"
+	exit 1
 fi
 
-/wigwag/system/bin/launch-edgenet.sh
-if [ $? -ne 0 ]; then
-    echo "Unable to create edgenet docker network"
-    exit 2
-fi
+RCONF=EDGE_RUN/coredns/resolv.conf
+
+write_resolv_conf(){
+	usage (){
+		echo "Constructs a resolv.conf file used by kubelet"
+		echo "  $0 <NODEIP>"
+		echo "Example"
+		echo "  $0 10.0.2.15"
+		exit
+	}
+	
+	if ! [ "$(id -u)" -eq 0 ]; then
+		echo "Must be run as root"
+		exit
+	fi
+
+	if [ $# -ne 1 ]; then
+		usage
+	fi
+	NODEIP="$1"
+	searchLine="hostname.local "
+	
+	grabSearch(){
+		local file="$1"
+		if [[ -e "$file" ]]; then
+			searchLine+="$(cat $file |  egrep ^search | sed 's/[^ ]* *//' )"
+		fi
+	}
+
+	addNS(){
+		local file="$1"
+		if [[ -e $file ]]; then
+			cat $file | grep nameserver | head -1 >> $RCONF
+		fi
+		grabSearch "$file"
+	}
+
+	write(){
+		if [[ ! -e EDGE_RUN/coredns ]]; then
+			mkdir -p EDGE_RUN/coredns
+		fi
+
+		echo "nameserver ${NODEIP}" > $RCONF
+		addNS cat/etc/resolv.conf
+		addNS /etc/resolv-conf.NetworkManager
+		echo "search $searchLine" >> $RCONF
+	}
+	write
+}
+
+
+
+write_resolv_conf EDGE_NODEIP
 
-exec /wigwag/system/bin/kubelet \
---root-dir=/var/lib/kubelet \
---offline-cache-path=/wigwag/system/var/lib/kubelet/store \
+exec EDGE_BIN/kubelet \
+--v=4 \
```

Comment:
I think logging verbosity 4 is too much for production, I suggest level 2 `--v=2` as used in Snap https://github.com/PelionIoT/snap-pelion-edge/blob/master/files/kubelet/scripts/launch-kubelet.sh#L79

It looks like you previously didn't provide any logging level, which defaults to value 0. This setting pretty much only reports Warnings, Errors, and minimal offline cache info. Level 2 gives us slightly more information in case we need to debug something on the gateway, while still not being too verbose.

Reply:
changed to -v2